smriprep/cli/run.py

```diff
@@ -294,38 +278,23 @@ def build_workflow(opts, retval):
     from nipype import logging, config as ncfg
     from ..__about__ import __version__
     from ..workflows.base import init_smriprep_wf
+    from ..utils import Spaces
     from niworkflows.utils.bids import collect_participants
 
-    # Set the default template to 'MNI152NLin2009c'
-    output_spaces = opts.output_spaces or OrderedDict([('MNI152NLin2009cAsym', {})])
+    spaces = Spaces(output=opts.output_spaces)
+    target_spaces = spaces.unique()
+    if not target_spaces:
+        # no output spaces have been set - should this blow up?
+        # Set the default template to 'MNI152NLin2009c'
+        spaces.add_space('MNI152NLin2009cAsym', output=False)
 
-    if opts.template:
-        print("""\
-The ``--template`` option has been deprecated in version 1.1.2. Your selected template \
-"%s" will be inserted at the front of the ``--output-spaces`` argument list. Please update \
-your scripts to use ``--output-spaces``.""" % opts.template, file=sys.stderr)
-        deprecated_tpl_arg = [(opts.template, {})]
-        # If output_spaces is not set, just replate the default - append otherwise
-        if opts.output_spaces is not None:
-            deprecated_tpl_arg += list(output_spaces.items())
-        output_spaces = OrderedDict(deprecated_tpl_arg)
-
-    if opts.fs_output_spaces:
-        print("""\
-The ``--fs-output-spaces`` option has been deprecated in version 1.1.2. Your selected output \
-surfaces "%s" will be appended to the ``--output-spaces`` argument list. Please update \
-your scripts to use ``--output-spaces``.""" % ', '.join(opts.fs_output_spaces), file=sys.stderr)
-        for fs_space in opts.fs_output_spaces:
-            if fs_space not in output_spaces:
-                output_spaces[fs_space] = {}
-
-    FS_SPACES = set(['fsnative', 'fsaverage', 'fsaverage6', 'fsaverage5'])
-    if opts.run_reconall and not list(FS_SPACES.intersection(output_spaces.keys())):
+    FS_SPACES = {'fsnative', 'fsaverage', 'fsaverage6', 'fsaverage5'}
+    if opts.run_reconall and not FS_SPACES.intersection(target_spaces):
         print("""\
 Although ``--fs-no-reconall`` was not set, no FreeSurfer output space (valid values are: \
 %s) was selected. Adding default "fsaverage5" to the \
 list of output spaces.""" % ', '.join(FS_SPACES), file=sys.stderr)
-        output_spaces['fsaverage5'] = {}
+        spaces.add_space('fsaverage5', output=False)
```

Comment:
Okay, you were right we have a problem here with my proposal. I'll mull over it tonight - I kind of like your approach.